### DISCUSSION TITLE
RE: UI Setup

### DISCUSSION THREAD

**Discussion by Pontiac76** [ZuluIDE/ZuluIDE-firmware#74] [Q&A]
I've been looking around for an IDE CDRom emulator for my 486 and P3 machine.  The only reason I haven't pulled the trigger is because I found out about ZuluID around 20 minutes ago, and the lack of a mechanism to show me the ISO selected and how to navigate to select the drive was off putting.  Then I ran across #63 and all of a sudden was REALLY interested.

Looking at the board, I'd assume we'd be plugging into this bank of to-be-populated headers?  Are there any photos of the potential setup, knowing full well that this is just pretty much a POC at this point?

![image](https://github.com/rabbitholecomputing/ZuluIDE-firmware/assets/4532020/c1b9bb2c-bb40-40b1-bd3b-29020f343ca8)

**Comment by aperezbios** [comment]
@Pontiac76 there's a brief demonstration video of the current state of things at https://www.zuluide.com/ZuluIDE-Control-UI-demo-12MB.mp4

I'll be AFK on a camping trip this weekend, but @hoffman373 could probably answer any questions you might have about the UI, from a software/firmware perspective.

**Reply by Pontiac76** [reply]
I'm not in TOO much of a rush for the devices at the moment.  I would like them yesterday, but, at the moment, trying to find out why my XTIDE isn't doing so well with my choice of hardware in my 486.  I like what I see, especially as a POC.  I "kinda" like what the Gotek has for its encoder manipulation, and this follows it quite nicely.

**Comment by hoffman373** [comment]
Hi @Pontiac76, we are working on a UI. The main code branch has support for a hardware UI. It has a small screen, rotary encoder, and buttons. It connects via the I2C header you mentioned in your post.

Prototype hard UI boards exist, but I @aperezbios will know more about current availability.

There is also a basic web service and accompanying basic HTML interface being worked on. This interface connects via I2C to an independent Raspberry PI pico. The web interface portion is functional, I am still working on the basic HTML interface and writing some instructions so people know how to set it up. Once those things are done it will be available for testing.

Both interfaces currently support CD-ROM devices only. The other emulated device types (e.g., generic ATAPI) will be supported soon.

Also, you can run the hardware and web service based UIs simultaneously.

Feel free to ask questions!

**Reply by Pontiac76** [reply]
Going from POC/WIP type of board, is there the expectation that if I were to order the boards today that something in the near future, prior to going live production, things are going to change significantly and what I bought becomes too obsolete and I'd have to obtain another UI board?
